core/packages/contracts/contracts/Vault.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// TODO: transfer ownership from deployer to SovereignTreasury
+// This contract actually holds Ether balances for each sovereignID.
+contract Vault is Ownable {
+    using Address for address payable;
+
+    event Deposited(bytes32 indexed sovereignID, uint256 amount);
```

Comment:
In my vault implementation, I store `msg.sender` in Deposited/Withdraw events as opposed to sovereignId. The implementation emits Lock/Unlock events from NativeTokens contract which contains the sovereignId or origin of the request. This allows us to see which version of NativeTokens contract initiated the withdraw/deposit. After an upgrade of `SovereignTreasury` contract the `msg.sender` should change to reflect the new version of `SovereignTreasury` in these events.

Reply:
Nice, adding `msg.sender` sounds like a good idea for audits.

I still think we should keep the `sovereignID` though. The deposit happens on behalf of a `sovereignID`, and is recorded in the `balances` mapping.

Reply:
`sovereignId` can be logged in events emitted from `SovereignTreasury.sol`.